engine/src/main/java/org/destinationsol/world/WorldBuilder.java

```diff
@@ -0,0 +1,119 @@
+ * all Generator classes and then initiates each SolSystem's build process. Two types of world generation classes are
+ * retrieved: those that subclass SolSystemGenerator and those at subclass FeatureGenerator.
+ */
+public class WorldBuilder {
+    /**
+     * These arrays hold instances of any class which extends SolSystemGenerator or FeatureGenerator, respectively.
+     */
+    private ArrayList<SolSystemGenerator> solSystemGenerators = new ArrayList<>();
+    private ArrayList<FeatureGenerator> featureGenerators = new ArrayList<>();
+    private Context context;
+    private final int numberOfSystems;
+    private int systemsBuilt; //This field is for testing purposes
+
+    public WorldBuilder(Context context, int numSystems) {
+        this.context = context;
+        numberOfSystems = numSystems;
+        populateSolSystemGeneratorList();
+        populateFeatureGeneratorList();
+        systemsBuilt = 0;
+    }
+
+    /**
+     * This method uses reflection to retrieve all SolSystemGenerator classes. They are added to the list
+     * of SolSystemGenerators.
+     */
+    private void populateSolSystemGeneratorList() {
+
+        for (Class generator : context.get(ModuleManager.class).getEnvironment().getSubtypesOf(SolSystemGenerator.class)) {
+            try {
+                SolSystemGenerator solSystemGenerator = (SolSystemGenerator) generator.newInstance();
+                solSystemGenerators.add(solSystemGenerator);
+            } catch (InstantiationException | IllegalAccessException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * This method uses reflection to retrieve all concrete FeatureGenerator classes. They are added to the list
+     * of FeatureGenerators.
+     */
+    private void populateFeatureGeneratorList() {
+        for (Class generator : context.get(ModuleManager.class).getEnvironment().getSubtypesOf(FeatureGenerator.class)) {
+            if (!Modifier.isAbstract(generator.getModifiers())) {
+                try {
+                    FeatureGenerator featureGenerator = (FeatureGenerator) generator.newInstance();
+                    featureGenerators.add(featureGenerator);
+                } catch (InstantiationException | IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    /**
+     * This method initializes the SolSystemGenerators. How many generators are initialized depends on the number
+     * of SolSystems the world is set to have. When there are multiple types of SolSystemGenerators available, this
+     * method chooses randomly from the list of all generators to decide which to create.
+     *
+     * This method calls setFeatureGenerators() on each SolSystemGenerator to give each SolSystem access to other
+     * FeatureGenerators (Planets, Mazes, etc.) so they can be initialized and incorporated into the SolSystem.
+     */
+    private void initializeSolSystemGenerators() {
```

Comment:
I think the discussion in Discord about random vs selected generators is worthwhile, but shouldn't hold back the basic worldgen structure from progressing. If you rename this something like `initializeSolarSystemGeneratorsUsingRandomGenerators()` and rename `build()` to `buildUsingRandomGenerators()`, that leaves open the possibility of using selected generators without needing to code a specific implementation of that idea. Lmk if that makes sense to you

Reply:
@IsaacLic I think the change you are suggesting makes sense (though I might think `initializeRandomSolarSystemGenerators()` might by a little less wordy. What do you think?)

I do wonder what this would look like in `SolGame`?  I suppose that from the UI there might be a string passed in that would contain the world-gen packages the use wants to enable. If the string is empty, then call `buildUsingRandomGenerators()`. If not, then call `buildUsingPackageGenerators(String s)`

Reply:
Sounds good to me :)